src/particles/distribution/Thermal.H

```diff
@@ -330,8 +330,8 @@ namespace distribution
         finalize ()
         {
             // deallocate
-            ThermalData::m_d_cdf1.clear();
-            ThermalData::m_d_cdf2.clear();
+            ThermalData::m_d_cdf1.release();
```

Comment:
As long as the `amrex::Gpu::DeviceVector` destructor is not called, there seems to be some arena memory left. This fixes the same shutdown issue as for the elements with array data.

Reply:
Will move this to a separate PR, because it seems to need more thought/testing.